projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/tabs/dossier-management-list-columns/dossier-management-list-columns.component.ts

```diff
@@ -14,15 +14,130 @@
+      viewType: 'string',
+      sortable: false,
+      propertyName: 'path',
+      translationKey: 'path',
+    },
+    {
+      viewType: 'string',
+      sortable: false,
+      propertyName: 'displayType',
+      translationKey: 'displayType',
+    },
+    {
+      viewType: 'string',
+      sortable: false,
+      propertyName: 'displayTypeParameters',
+      translationKey: 'displayTypeParameters',
+    },
+    {
+      viewType: 'string',
+      sortable: false,
+      propertyName: 'sortable',
+      translationKey: 'sortable',
+    },
+    {
+      viewType: 'string',
+      sortable: false,
+      propertyName: 'defaultSort',
+      translationKey: 'defaultSort',
+    },
+  ];
 
-  ngOnInit(): void {}
+  readonly fields$: Observable<Array<ListField>> = this.translateService.stream('key').pipe(
+    map(() =>
+      this.COLUMNS.map(column => ({
+        key: column.propertyName,
+        label: this.translateService.instant(`listColumn.${column.translationKey}`),
+        sortable: column.sortable,
+        ...(column.viewType && {viewType: column.viewType}),
+        ...(column.enum && {enum: column.enum}),
+      }))
+    )
+  );
+
+  readonly documentDefinitionName$: Observable<string> = this.route.params.pipe(
+    map(params => params.name || ''),
+    filter(docDefName => !!docDefName)
+  );
+
+  readonly caseListColumns$: Observable<Array<CaseListColumnView>> = this.translateService
+    .stream('key')
+    .pipe(
+      switchMap(() => this.documentDefinitionName$),
+      switchMap(documentDefinitionName => this.documentService.getCaseList(documentDefinitionName)),
+      map(columns =>
+        columns.map(column => ({
+          ...column,
+          title: column.title || '-',
+          sortable: column.sortable
+            ? this.translateService.instant('listColumn.sortableYes')
+            : this.translateService.instant('listColumn.sortableNo'),
+          defaultSort:
+            (column.defaultSort === 'ASC' &&
+              this.translateService.instant('listColumn.sortableAsc')) ||
+            (column.defaultSort === 'DESC' &&
+              this.translateService.instant('listColumn.sortableDesc')) ||
+            '-',
+          displayType: this.translateService.instant(
+            `listColumnDisplayType.${column?.displayType?.type}`
+          ),
+          displayTypeParameters: this.getDisplayTypeParametersView(
+            column.displayType.displayTypeParameters
+          ),
+        }))
+      )
+    );
+
+  constructor(
+    private readonly documentService: DocumentService,
+    private readonly route: ActivatedRoute,
+    private readonly translateService: TranslateService
+  ) {}
+
+  private getDisplayTypeParametersView(displayTypeParameters: DisplayTypeParameters): string {
+    if (displayTypeParameters?.dateFormat) {
+      return displayTypeParameters.dateFormat;
+    } else if (displayTypeParameters?.enum) {
+      return Object.keys(displayTypeParameters.enum).reduce((acc, curr) => {
```

Comment:
Just gonna write what we discussed earlier here, don't think it should prevent this PR from getting merged: If there are a lot of parameters here, a row can take up a lot of extra vertical space. I think we need some kind of overflow here, where anything larger than a certain size gets cut off, and is only shown when hovering over it.